File: CS/PersistentObjects/CustomerOrders.cs
using System;
using System.Collections.Generic;
using System.Text;
using DevExpress.Xpo;

namespace PersistentObjects {
    [NonPersistent]
    public class MyBaseObject : XPObject {
        public MyBaseObject(Session session) : base(session) { }
    }

    public class Customer : MyBaseObject  {
        public Customer(Session session) : base(session) { }

        string _CompanyName;
        public string CompanyName {
            get { return _CompanyName; }
            set { SetPropertyValue("CompanyName", ref _CompanyName, value); }
        }

        string _ContactName;
        public string ContactName {
            get { return _ContactName; }
            set { SetPropertyValue("ContactName", ref _ContactName, value); }
        }

        string _Country;
        public string Country {
            get { return _Country; }
            set { SetPropertyValue("Country", ref _Country, value); }
        }

        string _Phone;
        public string Phone {
            get { return _Phone; }
            set { SetPropertyValue("Phone", ref _Phone, value); }
        }

        [Association("Customer-Orders")]
        public XPCollection<Order> Orders {
            get { return GetCollection<Order>("Orders"); }
        }
    }

    public class Order : MyBaseObject {
        public Order(Session session) : base(session) { }

        DateTime _OrderDate;
        public DateTime OrderDate {
            get { return _OrderDate; }
            set { SetPropertyValue("OrderDate", ref _OrderDate, value); }
        }

        decimal _PaidTotal;
        public decimal PaidTotal {
            get { return _PaidTotal; }
            set { SetPropertyValue("PaidTotal", ref _PaidTotal, value); }
        }

        private Customer _Customer;

        [Association("Customer-Orders")]
        public Customer Customer {
            get {
                return _Customer;
            }
            set {
                SetPropertyValue("Customer", ref _Customer, value);
            }
        }
    }
}

File: VB/PersistentObjects/CustomerOrders.vb
Imports System
Imports System.Collections.Generic
Imports System.Text
Imports DevExpress.Xpo

Namespace PersistentObjects
	<NonPersistent>
	Public Class MyBaseObject
		Inherits XPObject

		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
	End Class

	Public Class Customer
		Inherits MyBaseObject

		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub

		Private _CompanyName As String
		Public Property CompanyName() As String
			Get
				Return _CompanyName
			End Get
			Set(ByVal value As String)
				SetPropertyValue("CompanyName", _CompanyName, value)
			End Set
		End Property

		Private _ContactName As String
		Public Property ContactName() As String
			Get
				Return _ContactName
			End Get
			Set(ByVal value As String)
				SetPropertyValue("ContactName", _ContactName, value)
			End Set
		End Property

		Private _Country As String
		Public Property Country() As String
			Get
				Return _Country
			End Get
			Set(ByVal value As String)
				SetPropertyValue("Country", _Country, value)
			End Set
		End Property

		Private _Phone As String
		Public Property Phone() As String
			Get
				Return _Phone
			End Get
			Set(ByVal value As String)
				SetPropertyValue("Phone", _Phone, value)
			End Set
		End Property

		<Association("Customer-Orders")>
		Public ReadOnly Property Orders() As XPCollection(Of Order)
			Get
				Return GetCollection(Of Order)("Orders")
			End Get
		End Property
	End Class

	Public Class Order
		Inherits MyBaseObject

		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub

		Private _OrderDate As Date
		Public Property OrderDate() As Date
			Get
				Return _OrderDate
			End Get
			Set(ByVal value As Date)
				SetPropertyValue("OrderDate", _OrderDate, value)
			End Set
		End Property

		Private _PaidTotal As Decimal
		Public Property PaidTotal() As Decimal
			Get
				Return _PaidTotal
			End Get
			Set(ByVal value As Decimal)
				SetPropertyValue("PaidTotal", _PaidTotal, value)
			End Set
		End Property

		Private _Customer As Customer

		<Association("Customer-Orders")>
		Public Property Customer() As Customer
			Get
				Return _Customer
			End Get
			Set(ByVal value As Customer)
				SetPropertyValue("Customer", _Customer, value)
			End Set
		End Property
	End Class
End Namespace
